### ISSUE TITLE
[Suggestion] Multithreading the screenshot action on versions before 1.13.1

### ISSUE THREAD

**Issue by CozyMeli-gitv2** [sp614x/optifine#7087]
## This suggestion applies to all versions below 1.13.1, however I am specifically going to use 1.12.2 as my example due to the fact I and many others play 1.12.2 optifine on the daily.

On all versions before 1.13.1, the screenshot action is handled on the **MAIN** thread, which means if you have a higher resolution monitor, even with a fast nvme ssd, you will have lag spikes when taking a screenshot, and if you have a lower end computer, you can expect lag spikes in the seconds long.

I am suggesting multithreading this action via the Optifine mod, to make a better experience for 1.12.2 players. This is a small change that would make 1.12.2 optifine users very happy. For the other versions, I would understand if this doesn't get considered, because there is a lot of them, and there is minimal reason to support unused old versions, but 1.12.2 and 1.8.9 should hopefully be considered as an exception. 

Here is the Mojang bug report for this on the official website. That includes all details that you may need to resolve this issue. https://bugs.mojang.com/browse/MC-33383

Thank you for reading and considering this patch. I would greatly appreciate this being put into the mod.

**Comment by CozyMeli-gitv2** [comment]
I disagree. Optifine focuses on performance, and after all is a performance enhancing mod. And Optifine has the screenshot resolution multiplier setting, which if you turn up, your game sometimes crashes if you take a screenshot or two. So as you can see, I would say at the bare minimum just adding it would fix a maxed out resolution setting from killing your instance.

**Comment by Janmm14** [comment]
For me, this is definitely in scope for Optifine. Bump!
